assets/components/paymentButtons/directDebitPopUpButton/directDebitPopUpButton.jsx

```diff
@@ -40,23 +40,23 @@ function mapDispatchToProps(dispatch) {
 
 const DirectDebitPopUpButton = (props: PropTypes) => {
 
-  let content = null;
+  const button = (
+    <button
+      id="qa-pay-with-direct-debit"
+      className="component-direct-debit-pop-up-button"
+      onClick={props.openDirectDebitPopUp}
+    >
+    Pay with direct debit
+    </button>);
 
   if (props.isPopUpOpen) {
-    content = <DirectDebitPopUpForm callback={props.callback} />;
-  } else {
-    content = (
-      <button
-        id="qa-pay-with-direct-debit"
-        className="component-direct-debit-pop-up-button"
-        onClick={props.openDirectDebitPopUp}
-      >
-        Pay with direct debit
-      </button>
-    );
+    return (
```

Comment:
Probably this is a picky detail, but having two returns in the function probably is more hazy than the previous version.

Reply:
But with the previous version the button disappeared after you clicked on it! This fixes that.

Reply:
I think it is the same logic, only in your case you are not using the `else` for modifying the flow of the code but returning directly if the condition is fulfilled.